src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,895 @@
+		{
+			base.OnPointerExited(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPointerExited.", this));
+#endif // MFSI_DEBUG
+
+			bool parentIsSubMenu = false;
+			MenuFlyoutPresenter parentPresenter = GetParentMenuFlyoutPresenter();
+
+			if (parentPresenter != null)
+			{
+				parentIsSubMenu = parentPresenter.IsSubPresenter;
+			}
+
+			m_menuHelper.OnPointerExited(args, parentIsSubMenu);
+		}
+
+		// PointerPressed event handler that ensures the pressed state.
+
+		protected override void OnPointerPressed(PointerRoutedEventArgs args)
+		{
+			base.OnPointerPressed(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPointerPressed.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnPointerPressed(args);
+		}
+
+		// PointerReleased event handler that shows MenuFlyoutSubItem in
+		// case of touch input.
+
+		protected override void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			base.OnPointerReleased(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPointerReleased.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnPointerReleased(args);
+
+		}
+
+
+		protected override void OnGotFocus(RoutedEventArgs args)
+		{
+			base.OnGotFocus(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnGotFocus.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnGotFocus(args);
+		}
+
+
+		protected override void OnLostFocus(RoutedEventArgs args)
+		{
+			base.OnLostFocus(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnLostFocus.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnLostFocus(args);
+		}
+
+		// KeyDown event handler that handles the keyboard navigation between
+		// the menu items and shows the MenuFlyoutSubItem in case of hitting
+		// the enter or right arrow key.
+
+		protected override void OnKeyDown(KeyRoutedEventArgs args)
+		{
+			base.OnKeyDown(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyDown.", this));
+#endif // MFSI_DEBUG
+
+			bool handled = false;
+			handled = args.Handled;
+			bool shouldHandleEvent = false;
+
+			if (!handled)
+			{
+				MenuFlyoutPresenter spParentPresenter = GetParentMenuFlyoutPresenter();
+
+				if (spParentPresenter != null)
+				{
+					var key = VirtualKey.None;
+
+					key = args.Key;
+
+					// Navigate each item with the arrow down or up key
+					if (key == VirtualKey.Down || key == VirtualKey.Up)
+					{
+						spParentPresenter.HandleUpOrDownKey(key == VirtualKey.Down);
+						UpdateVisualState();
+
+						// If we handle the event here, it won't get handled in m_menuHelper.OnKeyDown,
+						// so we'll do that afterwards.
+						shouldHandleEvent = true;
+					}
+				}
+			}
+
+			m_menuHelper.OnKeyDown(args);
+			args.Handled = !!shouldHandleEvent;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '!' operator just once or not at all.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)